### PR TITLE
Use json instead of xml in distributed performance test

### PR DESCRIPTION
**Pull request by blindpirate**
### Context

This fixes https://github.com/gradle/gradle-private/issues/1359 .

Also partially fixes https://github.com/gradle/gradle-private/issues/1422 because of adding catch blocks to http operations.

### CODE REVIEW

File: buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy [hunk 1]
```diff
@@ -162,71 +162,107 @@ class DistributedPerformanceTest extends ReportGenerationPerformanceTest {
...
-    private static String xmlToString(xmlObject) {
-        if (xmlObject != null) {
-            try {
-                return XmlUtil.serialize(xmlObject)
-            } catch (e) {
-                // ignore errors
-            }
+    private CoordinatorBuild resolveCoordinatorBuild() {
+        if (buildId) {
+            Map response = httpGet(path: "builds/id:$buildId")
```
Comment (wolfs) [inline buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: Don't you need to add request content type `JSON` to this request?

Comment (wolfs) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: Then how do you make sure the server sends you json? Doing the same request in the old code did the same http request and that returned XML.

File: buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy [hunk 2]
```diff
@@ -370,6 +401,7 @@ class DistributedPerformanceTest extends ReportGenerationPerformanceTest {
         client = new RESTClient("$teamCityUrl/httpAuth/app/rest/9.1")
         client.auth.basic(teamCityUsername, teamCityPassword)
         client.headers['Origin'] = teamCityUrl
+        client.headers['Accept'] = ContentType.JSON.toString()
```
Comment (blindpirate) [inline buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: @wolfs I think setting this globally here should be fine.

File: buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy (hunk 1 above)
Comment (blindpirate) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: I set `Accept` globally when creating `RESTClient`.

File: buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy (hunk 2 above)
Comment (wolfs) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: Does this work for downloading the artifact? The junit xml is `xml`, right? Or what would be the response there? If it works, then everything is good.

Comment (blindpirate) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: The junit xml is `xml`, but we download the whole zip as binary so no issue here. Anyway, I'll make sure everything works on CI, including cancelling.

**Pull request merged**